tests/instances.spec.ts

```diff
@@ -3,10 +3,117 @@ import {
   createInstance,
   deleteInstance,
   randomInstanceName,
+  renameInstance,
 } from "./instance-helpers";
+import {
+  assertCode,
+  assertReadMode,
+  setCodeInput,
+  setInput,
+  setOption,
+} from "./configuration-helpers";
+import { TIMEOUT } from "./constants";
 
 test("instance create and remove", async ({ page }) => {
   const instance = randomInstanceName();
   await createInstance(page, instance);
   await deleteInstance(page, instance);
 });
+
+test("instance rename", async ({ page }) => {
+  const instance = randomInstanceName();
+  await createInstance(page, instance);
+
+  const newName = instance + "-rename";
+  await renameInstance(page, instance, newName);
+
+  await deleteInstance(page, newName);
+});
+
+test("instance configuration edit", async ({ page }) => {
```

Comment:
This is quite big, doing all the configs in just one test, I wonder if we should split it like for profiles?